source/distributed/tria.cc

```diff
@@ -1124,10 +1124,545 @@ namespace
+        previous_global_first_quadrant,
+        parallel_forest->mpicomm,
+        0,
+        dest_data_fix.data(),
+        src_data_fix.data(),
+        cumulative_sizes_fix.back());
+
+      // Release memory of previously packed data.
+      src_data_fix.clear();
+      src_data_fix.shrink_to_fit();
+
+      // TODO: for variable transfer
+      // allocate memory for transfer
+      // p4est transfer fixed_begin for transfer of var data sizes
+      // p4est transfer custom for transfer of var data
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::unpack_cellstatus(
+      std::vector<quadrant_cell_relation_t> &quad_cell_relations) const
+    {
+      Assert(cumulative_sizes_fix.size() > 0,
+             ExcMessage("No data has been packed!"));
+      Assert((quad_cell_relations.size() == 0) || (dest_data_fix.size() > 0),
+             ExcMessage("No data has been received!"));
+
+      // Size of CellStatus object that will be unpacked on each cell.
+      const size_t size = sizeof(
+        typename parallel::distributed::Triangulation<dim,
+                                                      spacedim>::CellStatus);
+
+      // Iterate over all cells and overwrite the CellStatus
+      // information from the transferred data.
+      // Proceed buffer iterator position to next cell after
+      // each iteration.
+      auto it_qcrel     = quad_cell_relations.begin();
+      auto cit_dest_fix = dest_data_fix.cbegin();
+      for (; it_qcrel != quad_cell_relations.end();
+           ++it_qcrel, cit_dest_fix += cumulative_sizes_fix.back())
+        {
+          auto &cell_status = std::get<1>(*it_qcrel);
+
+          cell_status =
+            Utilities::unpack<typename parallel::distributed::
+                                Triangulation<dim, spacedim>::CellStatus>(
+              cit_dest_fix,
+              cit_dest_fix + size,
+              /*allow_compression=*/false);
+        }
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::unpack_data(
+      const std::vector<quadrant_cell_relation_t> &quad_cell_relations,
+      const unsigned int                           handle,
+      const std::function<void(
+        const typename dealii::Triangulation<dim, spacedim>::cell_iterator &,
+        const typename dealii::Triangulation<dim, spacedim>::CellStatus &,
+        const boost::iterator_range<std::vector<char>::const_iterator>)>
+        &unpack_callback) const
+    {
+      Assert(cumulative_sizes_fix.size() > 0,
+             ExcMessage("No data has been packed!"));
+      Assert((quad_cell_relations.size() == 0) || (dest_data_fix.size() > 0),
+             ExcMessage("No data has been received!"));
+
+      // Calculate offset and size from cumulated sizes.
+      const size_t offset = cumulative_sizes_fix[handle];
+      const size_t size   = cumulative_sizes_fix[handle + 1] - offset;
```

Comment:
`cumulative_sizes_fix` stores `unsigned int` objects -- so these variables should probably do the same